### ISSUE TITLE
Consider peerDependencies for provider SDK to Pulumi Core SDK dependency in Node

### ISSUE THREAD

**Issue by t0yv0** [pulumi/ci-mgmt#1240]
Going from:

https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/package.json#L16C5-L16C38

To a peerDependency.

TBD. pro/con? Is it breaking (likely yes).

**Comment by t0yv0** [comment]
Prior art: https://github.com/pulumi/pulumi/issues/3090

**Comment by guineveresaenger** [comment]
What are peerDependencies: https://nodejs.org/en/blog/npm/peer-dependencies

Sounds like a big con would be that manual intervention would have to happen on npm install if the dependency doesn't exist?

**Comment by t0yv0** [comment]
Was just jotting this down from standup convo, I need to read up on peerDependencies. Some more context why this came up is that in https://github.com/pulumi/pulumi-aws-apigateway/pull/178 YARN linked two versions of '@pulumi/pulumi` into the provider build which made if fail with a very obscure error. We're going to solve this with `yarn deduplicate` but it was suggested that peerDependencies could be another way.

**Comment by t0yv0** [comment]
Yeah the article explains it well I think. The suggestion here would be to treat `@pulumi/aws` SDK as a "plugin" that requests to be used with a certain range of `@pulumi/pulumi` versions by a peer dependency but not actually depend on it directly to avoid grabbing its own copy. So if that were done, the issue with `pulumi-aws-apigateway` having two copies of `@pulumi/pulumi` would not have occurred.

However @flostadler mentioned that moving from a direct dependency to a peer dependency is a breaking change, though I've not figured out why yet.

**Comment by t0yv0** [comment]
New related failure: https://github.com/pulumi/pulumi-service/pull/24879
